app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -16,6 +17,32 @@ def self.default_global_content_proxy
         end
       end
 
+      def update_references_to_proxy
+        root_repos = RootRepository
+          .where(:http_proxy_policy => Katello::RootRepository::USE_SELECTED_HTTP_PROXY)
+          .where(:http_proxy_id => nil)
+
+        if self == HttpProxy.default_global_content_proxy
+          root_repos += RootRepository.with_global_proxy
+        end
```

Comment:
we made a helper method for these lines:  https://github.com/Katello/katello/blob/9e9228f4d29601efefe8e95b36ca60fe9f199978/app/models/katello/root_repository.rb#L101

Reply:
actually that doesn't handle the global case, i swear we had one, let me see

Reply:
actually this method looks extremely similar to update_repository_proxy_details below, can you somehow combine them or reuse most of the logic?

Reply:
Using the `with_selected_proxy(nil)` seems to yield different results, but I'm not sure why. Still looking into that. I do plan to try and consolidate some of the commonality between the different hooks.